src/raftstore/store/peer.rs

```diff
@@ -459,13 +462,40 @@ impl Peer {
             ready.hs.take();
         }
 
+        if let Some(ref soft_state) = ready.ss {
+            if soft_state.leader_id == raft::INVALID_ID {
+                if self.leader_missing_time.is_none() {
+                    self.leader_missing_time = Some(Instant::now())
+                }
+            } else if self.is_initialized() {
```

Comment:
As the comment mentioned, if leader_missing_time is set to `None` even for uninitialized peer, when the leader is missing before the peer is initialized, then there would be no ready event triggered, and this peer couldn't be gc anymore.

It's OK to consider peer in this situation as leader missing, since the leader missing timeout is triggered after a long time. The leader is "missing" due to it fails to do the replication job in a specified long time, rather than  fails to send heartbeats.